### PR TITLE
report running request status to HF during CC connection loss

### PR DESCRIPTION
**Pull request by nidhi0622**
Fixes https://github.com/Azure/cyclecloud-symphony/issues/42

### CODE REVIEW

File: hostfactory/host_provider/src/cyclecloud_provider.py [hunk 1]
```diff
@@ -868,7 +876,31 @@ def _deperecated_terminate_status(self, input_json):
...
+                                 machines = termination_request.get("machines", {})
+                                 if machines:
+                                    for machine_id, hostname in machines.items():
+                                        response_machines.append({"name": hostname,
+                                                    "status": MachineStates.deleting,
+                                                    "result": MachineResults.executing,
+                                                    "machineId": machine_id})  
+                            response["requests"].append(request) 
+                       return self.json_writer(response)
+
```
Comment (bwatrous) [inline hostfactory/host_provider/src/cyclecloud_provider.py:903]: Probably we can remove the outer try: except: now

We should catch ALL exceptions here.

File: hostfactory/host_provider/src/cyclecloud_provider.py [hunk 2]
```diff
@@ -1335,6 +1367,7 @@ def main(argv=sys.argv, json_writer=simple_json_writer):  # pragma: no cover
         logger.debug("Input: %s", json.dumps(input_json))
                 
         if cmd == "templates":
+            logger.info("Using azurecc version %s", version.get_version())
```
Comment (bwatrous) [inline hostfactory/host_provider/src/cyclecloud_provider.py:1370]: Great!   Let's also clean up the "Arguments" output lines to make them "BEGIN {operationId} {cmd} - ..." and "END {operationId} {cmd}"

File: hostfactory/host_provider/src/cyclecloud_provider.py [hunk 3]
```diff
@@ -868,7 +876,31 @@ def _deperecated_terminate_status(self, input_json):
                 
                 if machines_to_terminate:
                     logger.warning("Re-attempting termination of nodes %s", machines_to_terminate)
-                    self.cluster.terminate(machines_to_terminate)
+                    try:
```
Comment (bwatrous) [inline hostfactory/host_provider/src/cyclecloud_provider.py:879]: This needs a unit test, but looks correct.
The unit test will help document the expected output

File: hostfactory/host_provider/src/cyclecloud_provider.py [hunk 4]
```diff
@@ -859,58 +858,56 @@ def _deperecated_terminate_status(self, input_json):
...
+                        if termination_id in terminate_requests:
+                                termination_request = terminate_requests.get(termination_id)
+                                machines = termination_request.get("machines", {})
+                                if machines:
+                                   for machine_id, hostname in machines.items():
+                                      response_machines.append({"name": hostname,
+                                                "status": MachineStates.deleting,
+                                                "result": MachineResults.executing,
+                                                "machineId": machine_id})  
+                        response["requests"].append(request)
```
Comment (dougclayton) [inline hostfactory/host_provider/src/cyclecloud_provider.py:901]: This looks like it appends a response per termination id, but using the same machines list that gets appended for each termination id. I would have expected a new list per termination_id.

Comment (nidhi0622) [reply hostfactory/host_provider/src/cyclecloud_provider.py:901]: Yes that's a mistake good catch, I will initialize response_machines=[] inside the for loop. So there is appropriate machine list for each termination id.

File: hostfactory/host_provider/src/cyclecloud_provider.py (hunk 3 above)
Comment (nidhi0622) [reply hostfactory/host_provider/src/cyclecloud_provider.py]: updated the unit test

File: hostfactory/host_provider/src/cyclecloud_provider.py [hunk 5]
```diff
@@ -851,34 +858,58 @@ def _deperecated_terminate_status(self, input_json):
...
+                try:
                     self.cluster.terminate(machines_to_terminate)
-                    
-                for termination_id in termination_ids:
-                    if termination_id in terminate_requests:
-                        termination = terminate_requests[termination_id]
-                        termination["terminated"] = True
+                except Exception:
+                    # Send HF request status as running so it remembers the request
+                    logger.exception("Could not terminate machines %s due to an exception, reported status as running", machines_to_terminate)
```
Comment (dougclayton) [inline hostfactory/host_provider/src/cyclecloud_provider.py:882]: Do we need to log the exception anywhere or is it already logged?

**Review by bwatrous — APPROVED** [review]

**Pull request merged**